### DISCUSSION TITLE
LDM Notes for November 15th, 2023

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#7702] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2023/LDM-2023-11-15.md

## Agenda

- `params` improvements
- Nullability analysis of collection expressions
- Evaluation of implicit indexers in object initializers

**Comment by sab39** [comment]
Any consideration of whether the general aligning between collection expressions and `params` might potentially lead to allowing spread elements in `params` expanded form?

**Reply by CyrusNajmabadi** [reply]
@sab39 yes. We would definitely be open to considering that. It would be hard though as this is already a range expression in those contexts

**Reply by huoyaoyuan** [reply]
Will it be ambiguous with range expression then?

**Comment by iam3yal** [comment]
I'm glad that `params` is about to get some love. :)

**Comment by walczakb** [comment]
Is working on `params` improvements still worth the effort when the same (and more) can be achieved with collection expressions by adding a pair of brackets at the call site? I am not criticizing, just curious.

**Reply by KennethHoff** [reply]
I somewhat agree, but at the same time, `params` is already a thing and I don't think we should leave it hanging just because we have a different way (*Looks at expressions*). This would basically complete that feature - being a wrapper around collection expressions.

**Reply by CyrusNajmabadi** [reply]
Our general interest here is actually consistency and simplicity.  It makes the language simpler and easier (which def helps long term) when we can unify concepts. 

If we weren't doing anything with params, I wouldn't recommend us doing this. But we are definitely doing params-span. Given that, we might as well spend a little more time here and do the entire consistency work.
